### PR TITLE
GH-41699: [Python][Parquet] Implement to_dict method on SortingColumn

### LINKED ISSUE #41699: 'pyarrow._parquet.SortingColumn' object has no attribute 'to_dict'
### Describe the bug, including details regarding any error messages, version, and platform.

When a `SortingColumn` is present, the metadata of a `ParquetFile` can not be serialized with `to_dict()` because `SortingColumn` is missing this method.

```
import polars as pl
import pyarrow.parquet as pq

df = pl.DataFrame({"a": [1, 2], "b": [10, 11]})
fname = "tmp.parquet"
pq.write_table(
    df.to_arrow(),
    fname,
    sorting_columns=[pq.SortingColumn(0),],
)

pqf = pq.ParquetFile(fname)
print(pqf.metadata.row_group(0).sorting_columns[0])
print(pqf.metadata.to_dict())
```
results in :
```
SortingColumn(column_index=0, descending=False, nulls_first=False)
...

  File "pyarrow/_parquet.pyx", line 892, in pyarrow._parquet.FileMetaData.to_dict
  File "pyarrow/_parquet.pyx", line 790, in pyarrow._parquet.RowGroupMetaData.to_dict
AttributeError: 'pyarrow._parquet.SortingColumn' object has no attribute 'to_dict'
```

### Component(s)

Parquet, Python

### PR DESCRIPTION
**Pull request by tlm365**
### Rationale for this change
Resolves #41699 .

### What changes are included in this PR?
Add `to_dict` method and test case

### Are these changes tested?
Yes

### Are there any user-facing changes?
No

* GitHub Issue: #41699

### CODE REVIEW

**Comment by AlenkaF** [comment]
Thank you for contributing!
Can you also add a test for metadata of a ParquetFile (`ParquetFile.metadata.to_dict()`)?

**Comment by tlm365** [comment]
@AlenkaF Thanks for review! I have added it.

**Review by AlenkaF — APPROVED** [review]
Just one minor nit otherwise LGTM. Thanks!

File: python/pyarrow/_parquet.pyx [hunk 1]
```diff
@@ -705,6 +705,21 @@ cdef class SortingColumn:
         """Whether null values appear before valid values (bool)."""
         return self.nulls_first
 
+    def to_dict(self):
+        """
+        Get dictionary representation of the SortingColumn.
```
[the remark this answers is not in the document]
Comment (tlm365) [reply python/pyarrow/_parquet.pyx:710]: @AlenkaF It's done. TYSM.

**Comment by AlenkaF** [comment]
The benchmark regression seems to be momentary so we can ignore.

**Pull request merged**